playbooks/reboot_cluster.yml

```diff
@@ -1,48 +1,123 @@
 ---
-- hosts: all:!db:!vyatta-*
+- hosts: controller
   serial: 1
   tasks:
   - name: Reboot
     command: shutdown -r now
     async: 0
     poll: 0
-    ignore_errors: true
+    failed_when: false
 
   - name: Wait for SSH to come back
     wait_for: host={{ ansible_default_ipv4.address }}
               port={{ ansible_ssh_port|default(22) }}
               delay=15
     delegate_to: "{{ groups['controller']|first }}"
 
+  - name: cluster health check
+    command: /etc/sensu/plugins/percona-cluster-size.rb -d /root/.my.cnf --expected 3 --criticality critical
+    register: cstat
+    until: cstat | succeeded
+    retries: 5
+    run_once: true
+    delegate_to: "{{ item }}"
+    with_items: "{{ play_hosts }}"
+
 - hosts: db_arbiter
   serial: 1
   tasks:
-  - name: Ensure Garbd is up before rebooting database servers
+  - name: cluster health check
+    command: /etc/sensu/plugins/percona-cluster-size.rb -d /root/.my.cnf --expected 3 --criticality critical
+    register: cstat
+    until: cstat | succeeded
+    retries: 5
+    run_once: true
+    delegate_to: "{{ item }}"
+    with_items: "{{ play_hosts }}"
+
+  - name: reboot
+    command: shutdown -r now
+    async: 0
+    poll: 0
+    failed_when: false
+
+- hosts: compute:!controller:!db_arbiter
```

Comment:
since db_arbiter is a compute node, it would need the same tests you're doing here before continuing. Since the only difference between the two plays is checking the cluster health, which you just did in the controller loop, you should be able to get rid of the db_arbiter play and just let the db_arbiter holding compute node get rebooted with the rest of the compute nodes.

You can add a test for cluster status after the ssh check, and have a conditional for the host being the arbiter.